sql/core/src/test/scala/org/apache/spark/sql/DatasetUnpivotSuite.scala

```diff
@@ -535,6 +548,98 @@ class DatasetUnpivotSuite extends QueryTest
         "val"),
       longStructDataRows)
   }
+
+  test("unpivot with struct expressions") {
+    checkAnswer(
+      wideDataDs.unpivot(
+        Array($"id"),
+        Array(
+          struct($"str1".as("str"), $"int1".cast(LongType).as("long")).as("str-int"),
```

Comment:
Shall we remove this test? We will only allow attributes as id and value expressions.

Reply:
and this PR should focus on the SQL API, not DataFrame.

Reply:
I can move this into a separate Scala DataFrame PR. The Scala API does support expressions.

Reply:
Removed in 8ed5d460.

Reply:
Moved to #38019.